{
    "body": "`spack buildcache` tries to create a cache entry for non-installed root specs:\r\n\r\n### Steps to reproduce the issue\r\n\r\n```console\r\n$ spack env create my_env\r\n$ spack -e my_env install --only=dependencies perl\r\n$ spack -e my_env buildcache create -d tmp\r\n==> Error: perl does not match any installed packages.\r\n==> Error: use one of the matching specs above\r\n```\r\n\r\n### Error Message\r\n\r\n```console\r\n# spack --debug --stacktrace -e ci buildcache create -d tmp\r\nlib/spack/spack/cmd/__init__.py:122 ==> [2020-08-03-22:53:22.459456] Imported buildcache from built-in commands\r\nlib/spack/spack/cmd/__init__.py:122 ==> [2020-08-03-22:53:22.461071] Imported buildcache from built-in commands\r\nlib/spack/spack/config.py:805 ==> [2020-08-03-22:53:22.462045] Reading config file /opt/spack/var/spack/environments/ci/spack.yaml\r\nlib/spack/spack/config.py:805 ==> [2020-08-03-22:53:22.472989] Reading config file /opt/spack/etc/spack/defaults/mirrors.yaml\r\nlib/spack/spack/cmd/buildcache.py:361 ==> [2020-08-03-22:53:22.474214] Buildcache files will be output to file:///tmp/build_cache\r\nlib/spack/spack/cmd/buildcache.py:253 ==> [2020-08-03-22:53:22.474733] find_matching_specs: about to parse specs for {perl}\r\nlib/spack/spack/database.py:362 ==> [2020-08-03-22:53:22.475940] DATABASE LOCK TIMEOUT: 3s\r\nlib/spack/spack/database.py:366 ==> [2020-08-03-22:53:22.476053] PACKAGE LOCK TIMEOUT: No timeout\r\nlib/spack/spack/config.py:805 ==> [2020-08-03-22:53:22.477811] Reading config file /opt/spack/etc/spack/defaults/repos.yaml\r\nlib/spack/spack/config.py:805 ==> [2020-08-03-22:53:22.479661] Reading config file /etc/spack/repos.yaml\r\nlib/spack/spack/cmd/buildcache.py:268 ==> [2020-08-03-22:53:22.528319] Error: perl does not match any installed packages.\r\nlib/spack/spack/cmd/buildcache.py:273 ==> [2020-08-03-22:53:22.528420] Error: use one of the matching specs above\r\nTraceback (most recent call last):\r\n  File \"/opt/spack/lib/spack/spack/main.py\", line 762, in main\r\n    return _invoke_command(command, parser, args, unknown)\r\n  File \"/opt/spack/lib/spack/spack/main.py\", line 490, in _invoke_command\r\n    return_val = command(parser, args)\r\n  File \"/opt/spack/lib/spack/spack/cmd/buildcache.py\", line 775, in buildcache\r\n    args.func(args)\r\n  File \"/opt/spack/lib/spack/spack/cmd/buildcache.py\", line 459, in createtarball\r\n    rebuild_index=args.rebuild_index)\r\n  File \"/opt/spack/lib/spack/spack/cmd/buildcache.py\", line 363, in _createtarball\r\n    matches = find_matching_specs(pkgs, env=env)\r\n  File \"/opt/spack/lib/spack/spack/cmd/buildcache.py\", line 273, in find_matching_specs\r\n    tty.die('use one of the matching specs above')\r\n  File \"/opt/spack/lib/spack/llnl/util/tty/__init__.py\", line 239, in die\r\n    sys.exit(1)\r\nSystemExit: 1\r\n```\r\n\r\n### Information on your system\r\n\r\n* **Spack:** 0.15.3\r\n* **Python:** 3.6.9\r\n* **Platform:** linux-ubuntu18.04-zen2\r\n\r\n### Additional information\r\n\r\n- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform\r\n- [x] I have searched the issues of this repo and believe this is not a duplicate\r\n- [x] I have run the failing commands in debug mode and reported the output",
    "user": "haampie",
    "url": "https://api.github.com/repos/spack/spack/issues/17858",
    "updated_at": "2020-08-13 16:59:21",
    "created_at": "2020-08-03 22:54:21",
    "closed_at": "2020-08-13 16:59:21",
    "state": "closed",
    "title": "buildcache create does not work with uninstalled root specs of environment",
    "number": 17858,
    "milestone": null,
    "labels": [
        "bug",
        "triage"
    ],
    "id": 672389232,
    "html_url": "https://github.com/spack/spack/issues/17858",
    "assignees": [],
    "comments": 1
}